Core/Core/Quizzes/QuizDetails/ViewModel/QuizDetailsViewModel.swift

```diff
@@ -0,0 +1,126 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2022-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public class QuizDetailsViewModel: QuizDetailsViewModelProtocol {
+    @Environment(\.appEnvironment) private var env
+    @Published public private(set) var state: QuizDetailsViewModelState = .loading
+    @Published public private(set) var courseColor: UIColor?
+
+    public var title: String { NSLocalizedString("Quiz Details", comment: "") }
+    public var subtitle: String { course.first?.name ?? "" }
+    public var showSubmissions: Bool { course.first?.enrollments?.contains(where: { $0.isTeacher || $0.isTA }) == true }
+    public var assignmentSubmissionBreakdownViewModel: AssignmentSubmissionBreakdownViewModel?
+    public var quizSubmissionBreakdownViewModel: QuizSubmissionBreakdownViewModel?
+    public var assignmentDateSectionViewModel: AssignmentDateSectionViewModel?
+    public var quizDateSectionViewModel: QuizDateSectionViewModel?
+
+    public var quizTitle: String { quiz.first?.title ?? "" }
+    public var pointsPossibleText: String { quiz.first?.pointsPossibleText ?? "" }
+    public var published: Bool { quiz.first?.published ?? false }
+    public var quizDetailsHTML: String? { quiz.first?.details }
+    public var attributes: [QuizAttribute] {
+        guard let quiz = quiz.first else { return [] }
+        return QuizAttributes(quiz: quiz, assignment: assignment).attributes
+    }
```

Comment:
These should be `@Published`. If data changes in CoreData and the `state` is already `ready` the UI won't update. As an alternative maybe we could move these to a struct and pass it as an associated enum value only if the `state` is `ready`, in other states these variables are meaningless.

Reply:
published cannot apply to computed properties, so I introduced:
```
    @Published private var quiz: Quiz?
    @Published private var assignment: Assignment?
    @Published private var course: Course?
```

which should do the trick